connection_pool.go

```diff
@@ -194,8 +196,6 @@ func (pool *ConnectionPool) Ping(host HostAddress, timeout time.Duration) error
 
 // Close closes all connection
 func (pool *ConnectionPool) Close() {
-	pool.rwLock.Lock()
```

Comment:
It's seems to be un-safe to iterate list, the length of list maybe changed during iterate.
```go
idleLen := pool.idleConnectionQueue.Len()
for i := 0; i < idleLen; i++ {
		pool.idleConnectionQueue.Front().Value.(*connection).close()
		pool.idleConnectionQueue.Remove(pool.idleConnectionQueue.Front())
	}
```

Reply:
Other list iterations may also have the same problems.

Reply:
agreed